app/scripts/modules/core/pipeline/config/stages/wait/SkipWait.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+
+import {IExecution} from 'core/domain/IExecution';
+import {IExecutionStage} from 'core/domain/IExecutionStage';
+import {ButtonBusyIndicator} from 'core/forms/buttonBusyIndicator/ButtonBusyIndicator';
+import {Application} from 'core/application/application.model';
+import {confirmationModalService} from 'core/confirmationModal/confirmationModal.service';
+import {executionService} from 'core/delivery/service/execution.service';
+import {duration} from 'core/utils/timeFormatters';
+import {OrchestratedItemRunningTime} from 'core/delivery/executionGroup/execution/OrchestratedItemRunningTime';
+
+interface IProps {
+  execution: IExecution;
+  stage: IExecutionStage;
+  application: Application;
+}
+
+interface IState {
+  submitting: boolean;
+  remainingWait?: string;
+}
+
+export class SkipWait extends React.Component<IProps, IState> {
+  private runningTime: OrchestratedItemRunningTime;
```

Comment:
Even if it doesn't change, it can still go into state. The "rule" is _"if it can change and it affects the view, it goes in state"_, not _"if it does not change, it does not go into state"_.  I sort of keep the 'if it affects the view, it goes into state, whether it changes or not' idea.

However, that being said, runningTime isn't a property that affects the view, its an object that has a callback with a property that can affect the view (like an rx subscription) So I think your pattern is great!